File: backend/app/routers/schema_pack.py
from fastapi import APIRouter, Request

router = APIRouter(prefix="/schema-pack", tags=["schema-pack"])


@router.get("")
def get_schema_pack(request: Request):
    schema_pack = getattr(request.app.state, "schema_pack", None)
    if not schema_pack:
        return {"loaded": False, "schema_pack": {}}
    return {"loaded": True, "schema_pack": schema_pack}

File: integrations/mc-server/src/routes/schema_pack.rs
use axum::{routing::get, Json, Router};
use std::sync::Arc;

use crate::state::AppState;

pub fn router() -> Router<Arc<AppState>> {
    Router::new().route("/schema-pack", get(get_schema_pack))
}

async fn get_schema_pack() -> Json<serde_json::Value> {
    Json(serde_json::json!({"loaded": false, "schema_pack": {}}))
}
